prover/src/lib.rs

```diff
@@ -35,13 +35,14 @@ pub use prover::StarkProof;
 /// Returns an error if program execution or STARK proof generation fails for any reason.
 pub fn prove(
     program: &Program,
+    stack: StackInputs,
     inputs: &ProgramInputs,
     options: &ProofOptions,
```

Comment:
I'd change the signature here to:
```Rust
pub fn prove(
    program: &Program,
    stack_inputs: &StackInputs,
    advice_inputs: &ProgramInputs,
    options: &ProofOptions,
) -> Result<(ProgramOutputs, StarkProof), ExecutionError>
```
In the subsequent PR we can replace `advice_inputs` with `advice_provider` and also rename `ProgramOutputs` into `StackOutputs`.

Reply:
Done!